kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -488,83 +494,102 @@ define([
-                                    }
-                                    dropdown.trigger('change');
+                data.push(viewData);
+            } else if (model.value && !model.displayValue && ddOptions.exact_match_on) {
+                const searchResults = await fetchData(model.value);
+                // verify that we have an exact match
+                let match = null;
+                for (const result of searchResults) {
+                    if (
+                        model.value.toLowerCase() === result[ddOptions.exact_match_on].toLowerCase()
+                    ) {
+                        match = result;
+                        break;
+                    }
+                }
+                if (match) {
+                    data.push(Object.assign(match, { selected: true }));
+                    model.value = match.id;
+                } else {
+                    model.exactMatchError = true;
+                }
+            }
+
+            dropdown.select2({
+                allowClear: true,
+                disabled: config.isViewOnly,
+                templateResult: formatObjectDisplay,
+                templateSelection: selectionTemplate,
+                data,
+                ajax: {
+                    delay: 250,
+                    processResults: (data) => {
+                        // update the currently selected one if applicable
+                        const currentData = dropdown.select2('data')[0];
+                        if (currentData) {
+                            const updatedData = data.results.filter((item) => {
+                                return item.id === currentData.id;
+                            });
+                            if (updatedData.length) {
+                                for (const key of Object.keys(updatedData[0])) {
+                                    currentData[key] = updatedData[0][key];
                                 }
+                                dropdown.trigger('change');
                             }
-                            return data;
-                        },
-                        transport: function (params, success, failure) {
-                            return fetchData(params.data.term)
-                                .then((data) => {
-                                    success({ results: data });
-                                })
-                                .catch((err) => {
-                                    console.error(err);
-                                    failure(err);
-                                });
-                        },
+                        }
+                        return data;
                     },
-                    placeholder: {
-                        id: 'select an option',
+                    transport: function (params, success, failure) {
+                        model.exactMatchError = false;
```

Comment:
Whenever a user triggers a search, we should clear the error, as any result they select should be valid now.